sdk/java/pulumi/src/main/java/com/pulumi/test/TestResult.java

```diff
@@ -0,0 +1,114 @@
+package com.pulumi.test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.pulumi.core.Output;
+import com.pulumi.exceptions.RunException;
+import com.pulumi.resources.Resource;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * A test result information.
+ */
+public class TestResult {
+    private final int exitCode;
```

Comment:
Why do we need to test exit codes?

Reply:
Yes, after giving this a though, exit codes are special case that works only for the CLI use and are not very useful to the user anyway. Moving the exit codes out of runner and removing from test result seam like a good idea.
Thank you for spotting this inconsistency.